### PR TITLE
vtk: fix and improve on issue introduced in #11337

### PR DESCRIPTION
**Pull request by CL-Jeremy**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/linuxbrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
- [x] Have you included the output of `brew gist-logs <formula>` of the build failure if your PR fixes a build failure. Please quote the exact error message.

> luho@marvin:~$ brew install vtk
> /home/luho/.linuxbrew/Homebrew/Library/Homebrew/brew.sh: line 4: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8): No such file or directory
> ==> Downloading https://www.vtk.org/files/release/8.2/VTK-8.2.0.tar.gz
> Already downloaded: /home/luho/.cache/Homebrew/downloads/bfbea40437b17f2734e32acfebf51e9d00c1dd8a34d63ac7410c77fdb46883e9--VTK-8.2.0.tar.gz
> Error: An exception occurred within a child process:
> &nbsp;&nbsp;NameError: undefined local variable or method `python_executable' for #<Formulary::FormulaNamespaced0780f3b8bd268fdd6760383b0108a87::Vtk:0x0000000003986c48>

### CODE REVIEW

**Comment by CL-Jeremy** [comment]
Some screenshots for evidence.

The effect of `inreplace`'s seem to have been achieved already without any action within the formula:
Bildschirmfoto 2019-06-15 um 16 18 24
Bildschirmfoto 2019-06-15 um 16 18 20

`ldd` and `readelf -d` seem to confirm this:
Bildschirmfoto 2019-06-15 um 16 52 04
Bildschirmfoto 2019-06-15 um 16 53 21

**Comment by iMichka** [comment]
This is currently blocked by #11696 (qt), which we have not been able to build since a few months now ...

**Comment by CL-Jeremy** [comment]
> This is currently blocked by #11696 (qt), which we have not been able to build since a few months now ...

Yup. I used -qt-zlib to circumvent that issue locally, but only because I needed it working quite urgently.

**Comment by iMichka** [comment]
Ok, I did the same, and a bottle was uploaded. Can you please rebase this on master?

**Comment by CL-Jeremy** [comment]
Yay!

File: Formula/vtk.rb [hunk 1]
```diff
@@ -62,32 +70,22 @@ def install
...
-      odie "No libpythonX.Y.{dylib|a} file found!"
+      odie "No libpythonX.Y.{#{dylib}|a} file found!"
     end
 
     mkdir "build" do
       system "cmake", "..", *args
       system "make"
       system "make", "install"
     end
-
```
Comment (iMichka) [inline Formula/vtk.rb]: Please do not delete these lines: just put them between an `unless OS.mac? / end` block.

File: Formula/vtk.rb [hunk 2]
```diff
@@ -34,6 +34,14 @@ class Vtk < Formula
   end
 
   def install
+    dylib = OS.mac? ? "dylib" : "so"
+
+    python_executable = `which python3`.strip
```
Comment (iMichka) [inline Formula/vtk.rb]: I would like to not allow to link against any system Python. We will use brewed python 3 only.
Please use the `Formula["python"]` construction to access the needed paths.

File: Formula/vtk.rb (hunk 1 above)
Comment (CL-Jeremy) [reply Formula/vtk.rb]: No, that's not how it is meant to be done. They shouldn't be needed at all. This should compile well even on macOS. I'm gonna test it in a minute.

Comment (iMichka) [reply Formula/vtk.rb]: This code comes from 0bdf9c0e270. Looks like we wanted to change the full paths to the opt paths for vtk 8.1.0 in 2018.

Maybe this is not needed anymore. Thanks for investigating this.

Comment (CL-Jeremy) [reply Formula/vtk.rb]: I'm not very familiar with CircleCI here. Does the successful build right after the merge indicate a successful build of the `vtk` formula? I don't see anything mentioning `vtk` in the log.

Comment (CL-Jeremy) [reply Formula/vtk.rb]: Oh, I think I misunderstood something: I was thinking it affects only build time path references (compiling and linking), but it actually deals with `.cmake` files. I can see it works for `python` (obviously, due to my modification of code above) but not for `hdf5`. I'll see how this works out after reworking on the Python 3 path detection part.

**Comment by iMichka** [comment]
Superseded by https://github.com/Homebrew/homebrew-core/pull/43557 and #15178.

**Pull request closed without merging**